### PR TITLE
Cleanup of duplicate actions detected by check_whitelists linter

### PR DESCRIPTION
**Pull request by mgerstner**

### CODE REVIEW

File: tools/remove_duplicate_entries.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+                linenr += 1
+
+                if linenr not in lines_to_drop:
+                    lines.append(line)
+
+        with open(path, 'w') as fd:
+            fd.write(''.join(lines))
+
+
+main = DuplicateEntryRemover()
```
Comment (maltek) [inline tools/remove_duplicate_entries.py:65]: maybe do the `__name__ == '__main__'` dance here

File: tools/pkcommon.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+class ProfileEntry:
+
+    path = ""
+    line = ""
+    linenr = 0
+    action = ""
+    settings = tuple()
+
+
+def parseProfile(path, callback):
```
Comment (maltek) [inline tools/pkcommon.py:27]: I feel the code would be cleaner if you'd `yield` the individial entries instead of calling a callback function on them.

**Review by maltek — APPROVED** [review]

**Comment by mgerstner** [comment]
Thank you @maltek for all the suggestions. They all make sense to me. Please see the additional commit for the changes. I also added some additional output to make it clearer what happens.

When you're okay with the changes please tell me so and I rebase first before you merge.

File: tools/remove_duplicate_entries.py [hunk 3]
```diff
@@ -13,47 +13,58 @@ def __init__(self):
...
     def rewriteProfile(self, path, lines_to_drop):
 
         lines = []
 
         with open(path) as fd:
 
-            linenr = 0
-
-            for line in fd.readlines():
+            for nr, line in enumerate(fd.readlines()):
```
Comment (maltek) [inline tools/remove_duplicate_entries.py:65]: sorry, should have been more clear: `enumerate(.., start=1)`

File: tools/pkcommon.py [hunk 4]
```diff
@@ -24,7 +24,7 @@ class ProfileEntry:
     settings = tuple()
 
 
-def parseProfile(path, callback):
+def parseProfile(path):
     """Parses the profile found in @path and calls @callback with an instance
     of ProfileEntry as parameter for each parsed entry."""
```
Comment (maltek) [inline tools/pkcommon.py:29]: This comment still needs updating.

File: tools/add_polkit_action.py [hunk 5]
```diff
@@ -152,23 +152,24 @@ def findDuplicateActions(self):
...
-                ret = False
-
-
         for profile in PROFILES:
 
             path = getProfilePath(profile)
 
-            parseProfile(path, checkDuplicate)
+            for entry in parseProfile(path):
+                ret &= self.checkDuplicate(entry)
```
Comment (maltek) [inline tools/add_polkit_action.py:160]: The `&=` is cute, but skimming over the code it made me stop for a minute to figure out what's going on. (Bitwise AND on a boolean).

Maybe just move the contents of `checkDuplicate` into the loop instead.

File: tools/remove_duplicate_entries.py (hunk 3 above)
Comment (mgerstner) [reply tools/remove_duplicate_entries.py]: Ah, didn't know about the `start` parameter. Done.

File: tools/pkcommon.py (hunk 4 above)
Comment (mgerstner) [reply tools/pkcommon.py]: Done.

File: tools/add_polkit_action.py (hunk 5 above)
Comment (mgerstner) [reply tools/add_polkit_action.py]: I made a simple `if not` statement out of it.

**Comment by maltek** [comment]
@mgerstner Looking good now :+1:

**Comment by mgerstner** [comment]
All right I rebased, from my POV this can be merged now.

**Pull request merged**